### DISCUSSION TITLE
debug log and logging in 5.1

### DISCUSSION THREAD

**Discussion by eurobank** [mindstellar/shopclass#387] [Q&A] [answered]
Hello.

Just saw my log and it is huge, wasn't like that before. Because of extended logging data. So an undefined index that was 1 line of log, now it is 250 !!!!! lines.

While that may help in some cases, right now it makes the log file very hard to read and find issues.

Is there a way to adjust the level of logging?

**Comment by navjottomer** [comment]
Ideally, you shouldn't have enabled OSC_DEBUG, errors are made more verbose. But 250 lines? that something else. What are these about?

**Reply by eurobank** [reply]
Each log of an issue is huge. As i said it was 1 line (5 if there was a crash with the stack) and now it is huge. 

For example there are some undefined index. I always have enabled debug to a staging site. How else will i know what works and what not.

But right now, i can't read the log. It is huge, not helping at all. The previous way was way much better (like Wordpress).

**Comment by navjottomer** [comment]
By default if disable osc_debug than, it'll only show FATAL or Warning error, you don't want to have PHP NOTICES, DEPRECATE in your error logs. I make a few more commit towards this.

**Reply by eurobank** [reply]
So the logic is TOTALLY changed. Because debug false always mean NO LOG or Display at all. Now you say something different.

**Reply by navjottomer** [reply]
If you don't define osclass debug then it only throws E_CORE_ERROR | E_CORE_WARNING | E_COMPILE_ERROR | E_ERROR | E_WARNING | E_PARSE | E_USER_ERROR | E_USER_WARNING as it had always. It will display those errors or not that's up to your PHP configuration. If you're comfortable displaying errors to your visitor then it's your choice.

**Reply by eurobank** [reply]
You mean in ERROR.LOG, yes sure.

And NO, if you define nothing as debug, the error is displaying on the page directly !! Bad.

**Reply by eurobank** [reply]
@navjottomer The fact here is that you totally changed the logic of Debug log. Totally. Also i mentioned several times that i'm using debug.log because i want to do so, under some certain circumstances. And now i can't. I can' even help you here, since there is no way to parse a huge log file and find the issues.

Never say i want to display errors to users/visitors. I always LOG the issues in all cases.

**Comment by navjottomer** [comment]
BTW I am not sure what logic changed?
OSC_DEBUG = OFF no error handling from osclass, it'll throw only fatal, warning or parse error, osclass will not interfare if it will show or not, that's up to your PHP config.
OSC_DEBUG = ON it'll show all types of errors and displaying errors will be enabled, with the new class they'll be stylized for attention-grabbing. But login of it still be controlled by your PHP config
**In Addition to OSC_DEBUG:**
OSC_DEBUG_LOG = OFF no log writing to debug.log file,  it'll not prevent it to write those errors to file configured in PHP config.
OSC_DEBUG_LOG = ON log will be written to debug.log file

Edit: By ON/OFF means true/false

**Reply by eurobank** [reply]
Is it Debug ON / OFF or Debug FALSE / TRUE? 

5.1 changed all that because i DIDN'T touch the config.php and now my log is huge.

Not sure what you are talking about, telling me how to use debug options. You think i do not know?

anyways, not communicating with each other.

**Reply by navjottomer** [reply]
Don't make assumptions about what I want, what I changed with some bug, it is not helpful at all.

**Reply by eurobank** [reply]
Ok, then see the log for an issue:
[debug.log.5.1.log](https://github.com/mindstellar/Osclass/files/7213050/debug.log.5.1.log)

**Comment by navjottomer** [comment, marked as answer]
See Now:  https://github.com/mindstellar/Osclass/blob/develop/oc-includes/osclass/classes/logger/OsclassErrors.php

**Reply by eurobank** [reply]
Excellent, apparently a bug and not anything else (changes etc etc).
